xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -496,11 +496,9 @@ void CGUIDialogVideoInfo::OnSearch(std::string& strSearch)
       if (iItem < 0)
         return;
 
-      CFileItem* pSelItem = new CFileItem(*items[iItem]);
+      CFileItemPtr pSelItem(new CFileItem(*items[iItem]));
 
-      OnSearchItemFound(pSelItem);
-
-      delete pSelItem;
+      OnSearchItemFound(pSelItem.get());
```

Comment:
To drive it to the max, <code>OnSearchItemFound(&(items[iItem]));</code> also should work.